pkg/externalip/registry/externalip/strategy.go

```diff
@@ -0,0 +1,73 @@
+package externalip
+
+import (
+	kapi "k8s.io/kubernetes/pkg/api"
+	"k8s.io/kubernetes/pkg/fields"
+	"k8s.io/kubernetes/pkg/labels"
+	"k8s.io/kubernetes/pkg/registry/generic"
+	"k8s.io/kubernetes/pkg/runtime"
+	"k8s.io/kubernetes/pkg/util/validation/field"
+
+	"github.com/openshift/origin/pkg/externalip/api"
+	"github.com/openshift/origin/pkg/externalip/api/validation"
+)
+
+type externalIPStrategy struct {
+	runtime.ObjectTyper
+}
+
+// Strategy is the default logic that applies when creating and
+// updating ExternalIP objects via the REST API.
+var Strategy = externalIPStrategy{kapi.Scheme}
+
+func (externalIPStrategy) PrepareForUpdate(obj, old runtime.Object) {
+}
+
+func (externalIPStrategy) NamespaceScoped() bool {
+	return false
+}
+
+func (externalIPStrategy) GenerateName(base string) string {
+	return base
+}
+
+func (externalIPStrategy) PrepareForCreate(obj runtime.Object) {
+	externalIP := obj.(*api.ExternalIP)
+	// Limit to kind/namespace/name
+	externalIP.ServiceRef = kapi.ObjectReference{
+		Kind:      externalIP.ServiceRef.Kind,
+		Namespace: externalIP.ServiceRef.Namespace,
+		Name:      externalIP.ServiceRef.Name,
+	}
+}
+
+func (externalIPStrategy) Validate(ctx kapi.Context, obj runtime.Object) field.ErrorList {
+	externalIP := obj.(*api.ExternalIP)
+	return validation.ValidateExternalIP(externalIP)
+}
+
+func (externalIPStrategy) AllowCreateOnUpdate() bool {
+	return false
+}
+
+func (externalIPStrategy) AllowUnconditionalUpdate() bool {
+	return false
+}
+
+func (externalIPStrategy) Canonicalize(obj runtime.Object) {
+}
+
+func (externalIPStrategy) ValidateUpdate(ctx kapi.Context, obj, old runtime.Object) field.ErrorList {
+	// TODO?
```

Comment:
You need to at minimum validate object meta changes and disallow all other updates (up front).  See PetSet for an example.